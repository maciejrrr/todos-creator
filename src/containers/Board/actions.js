import { ADD_CARD } from './constants';

export const addCard = ({ card }) => ({
  type: ADD_CARD,
  card,
});
