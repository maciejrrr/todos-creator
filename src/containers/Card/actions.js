import { ADD_CARD, EDIT_CARD, ADD_TASK } from './constants';

export const addCard = ({ card }) => ({
  type: ADD_CARD,
  card,
});

export const editCard = ({ cardId, name }) => ({
  type: EDIT_CARD,
  cardId,
  name,
});

export const addTask = ({ task }) => ({
  type: ADD_TASK,
  task,
});
