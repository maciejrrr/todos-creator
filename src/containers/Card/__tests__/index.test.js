import { Card, mapDispatchToProps } from '../index';
import { addTask, editCard } from '../actions';
import { DragDropContext } from 'react-beautiful-dnd';

describe('Card component', () => {
  let props;
  beforeEach(() => {
    props = {
      tasks: [],
      addTask: jest.fn(),
      card: { id: 1, name: 'first card' },
    };
  });

  describe('snapshots', () => {
    describe('no tasks', () => {
      it('renders AddTask component without any tasks', () => {
        const tree = renderer
          .create(
            <DragDropContext>
              <Card {...props} />
            </DragDropContext>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('with tasks', () => {
      it('renders AddCard component and tasks', () => {
        const tasks = [{ id: 1, name: 'first task' }, { id: 2, name: 'second task' }];
        const tree = renderer
          .create(
            <DragDropContext>
              <Card {...props} tasks={tasks} />
            </DragDropContext>
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('mapDispatchToprops', () => {
    let dispatch;
    let result;
    beforeEach(() => {
      dispatch = jest.fn();
      result = mapDispatchToProps(dispatch);
    });

    describe('addTask', () => {
      it('dispatches addTask with task object', () => {
        const task = { id: 1, name: 'test' };
        result.addTask({ task });
        expect(dispatch).toBeCalledWith(addTask({ task }));
      });
    });

    describe('editCard', () => {
      it('dispatches editCard with task object', () => {
        const id = 1;
        const text = 'test text';
        result.editCard({ id, text });
        expect(dispatch).toBeCalledWith(editCard({ cardId: id, name: text }));
      });
    });
  });
});
