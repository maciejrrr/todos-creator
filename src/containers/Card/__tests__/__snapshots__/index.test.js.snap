// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component snapshots no tasks renders AddTask component without any tasks 1`] = `
<div
  className="sc-dnqmqq mseXG"
>
  <div
    className="sc-iwsKbI kSpZQy"
  >
    <p
      className="sc-gZMcBi fwEucS"
    >
      first card
    </p>
  </div>
  <div
    className="sc-bxivhb TDOcQ"
  >
    <a
      className="sc-EHOje iCMDbo"
      onClick={[Function]}
    >
      Add a task...
    </a>
  </div>
</div>
`;

exports[`Card component snapshots with tasks renders AddCard component and tasks 1`] = `
<div
  className="sc-dnqmqq mseXG"
>
  <div
    className="sc-iwsKbI kSpZQy"
  >
    <p
      className="sc-gZMcBi fwEucS"
    >
      first card
    </p>
    <div
      className="sc-gzVnrw comcWq"
    >
      <p
        className="sc-htoDjs kNRfwO"
      >
        first task
      </p>
    </div>
    <div
      className="sc-gzVnrw comcWq"
    >
      <p
        className="sc-htoDjs kNRfwO"
      >
        second task
      </p>
    </div>
  </div>
  <div
    className="sc-bxivhb TDOcQ"
  >
    <a
      className="sc-EHOje iCMDbo"
      onClick={[Function]}
    >
      Add a task...
    </a>
  </div>
</div>
`;
